roles/tackle/templates/deployment-ui.yml.j2

```diff
@@ -37,6 +35,51 @@ spec:
         role: {{ ui_service_name }}
     spec:
       containers:
+{% if feature_auth_required|bool and feature_auth_type == "oauth" %}
```

Comment:
Agreed the conditionals get crazy in the UI deployment , I'm wondering if perhaps there is away in jinja2 to pre-assemble or pre-process based on selected settings somewhere else and just dump the final contents in this deployment

Reply:
@fbladilo I was complaining earlier on in the thread about nested conditionals before I did cleanup in some of my prior rebases. Taking this for example:

```
{% if feature_auth_required|bool %}
{% if feature_auth_type == "oauth" %}
...
{% endif %}
{% endif %}
```

In some places they were three and four deep and very hard to follow.

If you see an obvious way to clean up further I'm happy to do so.

Reply:
I think having whole files that Ansible chooses can sometimes be nice if the conditionals are super nested and hard to follow?